crates/winch/src/compiler.rs

```diff
@@ -76,21 +76,33 @@ impl wasmtime_environ::Compiler for Compiler {
         ))
     }
 
-    fn compile_host_to_wasm_trampoline(
+    fn compile_array_to_wasm_trampoline(
         &self,
-        ty: &wasmtime_environ::WasmFuncType,
+        translation: &ModuleTranslation<'_>,
+        types: &ModuleTypes,
+        index: DefinedFuncIndex,
     ) -> Result<Box<dyn Any + Send>, CompileError> {
-        let wasm_ty = wasmparser::FuncType::new(
-            ty.params().iter().copied().map(Into::into),
-            ty.returns().iter().copied().map(Into::into),
-        );
+        let _ = (translation, types, index);
+        todo!()
```

Comment:
cc @saulecabrera we opted to leave these as `todo!()` in winch, but this effectively breaks compilation with winch in Wasmtime (and tests are additionally disabled lower in this PR).

Is that ok with you though?

Reply:
That's ok with me, thanks for the heads up. I'll work on the new version of the trampolines for winch once this PR lands.

Reply:
Ah yes, sorry I meant to tag you @saulecabrera but I forgot. Thanks for your help getting this PR landed, and willingness to work on follow ups!